### PR TITLE
Integrated-fixed-arwen-getCompiledCode

### PR DESCRIPTION
**Pull request by sasurobert**
New arwen release integrated.

### CODE REVIEW

File: cmd/node/config/config.toml [hunk 1]
```diff
@@ -580,7 +580,7 @@
 
 [VirtualMachine]
     [VirtualMachine.Execution]
-        OutOfProcessEnabled = true
+        OutOfProcessEnabled = false
```
Comment (SebastianMarian) [inline cmd/node/config/config.toml:583]: This would be reverted later?

Comment (sasurobert) [reply cmd/node/config/config.toml:583]: don't think so. the engine is save now to be used inprocess always

File: integrationTests/vm/arwen/delegation/delegation_test.go [hunk 2]
```diff
@@ -192,3 +200,156 @@ func addNodes(context *arwen.TestContext, stakePerNode int, numNodes int) {
...
+
+func delegationProcessManyTimes(t *testing.T, warmInstance bool, outOfProcess bool, txPerBenchmark int, numRun int) {
+	ownerAddressBytes := []byte("12345678901234567890123456789011")
+	ownerNonce := uint64(11)
+	ownerBalance := big.NewInt(10000000000000)
+	gasPrice := uint64(1)
+	gasLimit := uint64(10000000000)
+
+	scCode := arwen.GetSCCode("../testdata/delegation/delegation_v0_5_1_full.wasm")
+	// 17918321 - stake in active - 11208675 staking in waiting - 28276371 - unstake from active
```
Comment (andreibancioiu) [inline integrationTests/vm/arwen/delegation/delegation_test.go:236]: Does this comment still apply?

**Review by andreibancioiu — APPROVED** [review]

File: integrationTests/vm/arwen/delegation/delegation_test.go (hunk 2 above)
Comment (sasurobert) [reply integrationTests/vm/arwen/delegation/delegation_test.go:236]: yes. it is for testing purposes.

**Review by andreibancioiu — APPROVED** [review]

File: core/converters.go [hunk 3]
```diff
@@ -90,6 +90,20 @@ func GetShardIDString(shardID uint32) string {
 	return fmt.Sprintf("%d", shardID)
 }
 
+// ConvertShardIDToUint32 converts shard id from string to uint32
+func ConvertShardIDToUint32(shardIDStr string) (uint32, error) {
+	if shardIDStr == "metachain" {
+		return MetachainShardId, nil
+	}
+
+	shardID, err := strconv.ParseInt(shardIDStr, 10, 64)
```
Comment (andreibancioiu) [inline core/converters.go:99]: `ParseInt(shardIDStr, 10, 32)` - but it really does not matter in this case, actually.

File: process/smartContract/hooks/blockChainHook.go [hunk 4]
```diff
@@ -57,8 +64,11 @@ type BlockChainHookImpl struct {
 	mutCurrentHdr sync.RWMutex
 	currentHdr    data.HeaderHandler
 
-	compiledScStorage storage.Storer
 	compiledScPool    storage.Cacher
+	compiledScStorage storage.Storer
+	configSCStorage   config.StorageConfig
+	workingDir        string
+	nilStorage        bool
```
Comment (andreibancioiu) [inline process/smartContract/hooks/blockChainHook.go:71]: Perhaps use the same name here, `nilCompiledSCStore`?

**Review by andreibancioiu — APPROVED** [review]

File: core/converters.go (hunk 3 above)
Comment (sasurobert) [reply core/converters.go:99]: it was copy paste from existing code.

File: storage/lastestData/latestDataProvider.go [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package factory
+package lastestData
```
Comment (SebastianMarian) [inline storage/lastestData/latestDataProvider.go:1]: latestData instead lastestData. Also rename the folder

File: storage/lastestData/latestDataProvider_test.go [hunk 6]
```diff
@@ -1,4 +1,4 @@
-package factory
+package lastestData
```
Comment (SebastianMarian) [inline storage/lastestData/latestDataProvider_test.go:1]: latestData instead lastestData. Also rename the folder

File: process/smartContract/hooks/blockChainHook.go [hunk 7]
```diff
@@ -71,24 +81,26 @@ func NewBlockChainHookImpl(
...
+		marshalizer:      args.Marshalizer,
+		uint64Converter:  args.Uint64Converter,
+		builtInFunctions: args.BuiltInFunctions,
+		compiledScPool:   args.CompiledSCPool,
+		configSCStorage:  args.ConfigSCStorage,
+		workingDir:       args.WorkingDir,
+		nilStorage:       args.NilCompiledSCStore,
+	}
+
+	err = blockChainHookImpl.makeCompiledSCStorage()
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply process/smartContract/hooks/blockChainHook.go:98]: No. As first you need to create. Empty it and recreate it. You want a clean storage.

File: process/smartContract/hooks/blockChainHook.go [hunk 8]
```diff
@@ -544,10 +553,41 @@ func (bh *BlockChainHookImpl) ClearCompiledCodes() {
 	bh.compiledScPool.Clear()
 	err := bh.compiledScStorage.DestroyUnit()
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply process/smartContract/hooks/blockChainHook.go:554]: Nom destroy is the one cleaning the storage.

File: cmd/node/factory/structs.go [hunk 9]
```diff
@@ -2473,7 +2484,7 @@ func CreateLatestStorageDataProvider(
 ) (storage.LatestStorageDataProviderHandler, error) {
 	directoryReader := storageFactory.NewDirectoryReader()
 
-	latestStorageDataArgs := storageFactory.ArgsLatestDataProvider{
+	latestStorageDataArgs := lastestData.ArgsLatestDataProvider{
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply cmd/node/factory/structs.go:2487]: renamed

File: storage/lastestData/latestDataProvider.go (hunk 5 above)
Comment (sasurobert) [reply storage/lastestData/latestDataProvider.go:1]: renamed

File: storage/lastestData/latestDataProvider_test.go (hunk 6 above)
Comment (sasurobert) [reply storage/lastestData/latestDataProvider_test.go:1]: renamed

File: process/smartContract/hooks/blockChainHook.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (sasurobert) [reply process/smartContract/hooks/blockChainHook.go:71]: renamed

File: integrationTests/testProcessorNode.go [hunk 10]
```diff
@@ -1193,17 +1193,17 @@ func (tpn *TestProcessorNode) initMetaInnerProcessors() {
...
+		Accounts:           tpn.AccntState,
+		PubkeyConv:         TestAddressPubkeyConverter,
+		StorageService:     tpn.Storage,
+		BlockChain:         tpn.BlockChain,
+		ShardCoordinator:   tpn.ShardCoordinator,
+		Marshalizer:        TestMarshalizer,
+		Uint64Converter:    TestUint64Converter,
+		BuiltInFunctions:   builtInFuncs,
+		DataPool:           tpn.DataPool,
+		NilCompiledSCStore: true,
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply integrationTests/testProcessorNode.go:1205]: done

File: process/factory/metachain/vmContainerFactory_test.go [hunk 11]
```diff
@@ -25,16 +25,16 @@ func createMockVMAccountsArguments() hooks.ArgBlockChainHook {
...
-		CompiledSCPool:    datapool.SmartContracts(),
+		PubkeyConv:         mock.NewPubkeyConverterMock(32),
+		StorageService:     &mock.ChainStorerMock{},
+		BlockChain:         &mock.BlockChainMock{},
+		ShardCoordinator:   mock.NewOneShardCoordinatorMock(),
+		Marshalizer:        &mock.MarshalizerMock{},
+		Uint64Converter:    &mock.Uint64ByteSliceConverterMock{},
+		BuiltInFunctions:   builtInFunctions.NewBuiltInFunctionContainer(),
+		DataPool:           datapool,
+		NilCompiledSCStore: true,
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply process/factory/metachain/vmContainerFactory_test.go:36]: done

**Review by SebastianMarian — APPROVED** [review]

File: process/smartContract/hooks/blockChainHook.go (hunk 8 above)
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/smartContract/hooks/blockChainHook.go:554]: ok

File: process/smartContract/hooks/blockChainHook.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/smartContract/hooks/blockChainHook.go:98]: ok

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**